app/api/ada/adaTransactions/adaNewTransactions.js

```diff
@@ -199,7 +200,12 @@ function decodeRustTxWithInputs(
   const selectedInputs = availableInputs.filter((inp: TxInput) => set.has(pointerToStr(inp.ptr)));
   return {
     inputs: selectedInputs,
-    outputs: tx.tx.tx.outputs
+    outputs: tx.tx.tx.outputs.map(x => {
+      if (x.address === changeAddress.cadId) {
+        x.fullAddress = changeAddress;
```

Comment:
maybe we could use `isChangeAddress` instead of `fullAddress`? the code looks good! but I was thinking that if someone sees `fullAdress` is difficult to infer that this is used as to determinate a changeAddress.

Reply:
Makes sense, will change in a bit